src/search/SearchMessagesScreen.js

```diff
@@ -37,17 +38,17 @@ class SearchMessagesScreen extends PureComponent<Props, State> {
    *  Performs a network request associated with a query. Does not
    *  update or access internal state (except `auth`).
    */
-  performQueryRaw = async (query: string): Promise<Message[]> => {
+  performQueryRaw = async (query: string) => {
```

Comment:
Then, after those two refactors I mentioned (and before your original commit), you can share code for passing arguments to api.getMessages and messageFetchComplete, since it takes exactly the same four pieces of data, with the same names! This can go near the top of handleQueryChangeInner. You can call it, for example, fetchArgs:

```
const fetchArgs = {
  narrow: SEARCH_NARROW(query),
  anchor: LAST_MESSAGE_ANCHOR,
  numBefore: 20,
  numAfter: 0,
}
```

To make that sharing easy, let's remove the `performQueryRaw` method. It's not very large, and it only has one call site, so it's OK to run inline. Replace its call site with `const networkPromise = api.getMessages(auth, fetchArgs);`, using the fetchArgs you've just defined.

Then, in your original commit, when you call messageFetchComplete, you can spread fetchArgs.

```
dispatch(
  messageFetchComplete({
    ...args,
    messages: networkResponse.messages,
    foundNewest: networkResponse.found_newest,
    foundOldest: networkResponse.found_oldest,
  }),
);
```

I think it's much easier to read than before. It's easy to see that the same four pieces of data are used in api.getMessages and messageFetchComplete, without having to read each one of them and compare them. 🙂